File: app/src/main/java/com/msbte/modelanswerpaper/myadapter.java
package com.msbte.modelanswerpaper;

import android.app.Activity;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.firebase.ui.database.FirebaseRecyclerAdapter;
import com.firebase.ui.database.FirebaseRecyclerOptions;


public class myadapter extends FirebaseRecyclerAdapter<model,myadapter.myviewholder>
{

    public myadapter(@NonNull FirebaseRecyclerOptions<model> options) {
        super(options);
    }

    @Override
    protected void onBindViewHolder(@NonNull final myviewholder holder, int position, @NonNull final model model) {

        holder.name.setText(model.getFilename());

        holder.download.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent=new Intent(holder.download.getContext(),download.class);

                intent.putExtra("filename",model.getFilename());
                intent.putExtra("fileurl",model.getFileurl());

                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                holder.download.getContext().startActivity(intent);
            }
        });
    }

    @NonNull
    @Override
    public myviewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view= LayoutInflater.from(parent.getContext()).inflate(R.layout.candle_iteam,parent,false);
        return  new myviewholder(view);
    }

    public class myviewholder extends RecyclerView.ViewHolder
    {

        TextView name;
        Button download;

        public myviewholder(@NonNull View itemView)
        {
            super(itemView);

            name=itemView.findViewById(R.id.mpname);
            download=itemView.findViewById(R.id.downloadfile2);


        }
    }
}

File: app/src/main/java/com/msbte/modelanswerpaper/myadapter.kt
package com.msbte.modelanswerpaper

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.firebase.ui.database.FirebaseRecyclerOptions
import com.msbte.modelanswerpaper.myadapter.myviewholder

class myadapter(options: FirebaseRecyclerOptions<model?>) :
    FirebaseRecyclerAdapter<model, myviewholder>(options) {
    override fun onBindViewHolder(holder: myviewholder, position: Int, model: model) {
        holder.name.text = model.filename
        holder.download.setOnClickListener {
            val intent = Intent(holder.download.context, download::class.java)
            intent.putExtra("filename", model.filename)
            intent.putExtra("fileurl", model.fileurl)
            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
            holder.download.context.startActivity(intent)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): myviewholder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.candle_iteam, parent, false)
        return myviewholder(view)
    }

    inner class myviewholder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        lateinit var name: TextView
        lateinit var download: Button

        init {
            name = itemView.findViewById(R.id.mpname)
            download = itemView.findViewById(R.id.downloadfile2)
        }
    }
}